cogs/util/tmp_channel_util.py

```diff
@@ -0,0 +1,208 @@
+                      verified: PermissionOverwrite(view_channel=True, connect=True),
+                      moderator: PermissionOverwrite(view_channel=not is_moderator, connect=not is_moderator),
+                      guild.default_role: PermissionOverwrite(view_channel=False, connect=False)}
+
+        voice_channel: VoiceChannel = await guild.create_voice_channel(name=name,
+                                                                       category=channel_category,
+                                                                       overwrites=overwrites,
+                                                                       nsfw=False,
+                                                                       reason="")
+
+        entry: Union[GamingChannel, StudyChannel] = await db.insert_one(
+            (member, text_channel, voice_channel, TmpChannelUtil.create_token(), None))
+
+        try:
+            await member.move_to(voice_channel, reason="Create Tmp Channel")
+        except Exception:
+            pass
+
+        return entry
+
+    @staticmethod
+    async def update_category_and_voice_channel(value: int,
+                                                ctx: Context,
+                                                db: PrimitiveMongoData,
+                                                key: ConfigurationNameEnum,
+                                                msg: str):
+        """
+        Updates a db entry.
+        """
+        find = {key.value: {"$exists": True}}
+        if await db.find_one(find):
+            await db.update_one(find, {key.value: value})
+        else:
+            await db.insert_one({key.value: value})
+        embed = Embed(title="Tmp channels",
+                      description=f"Set {msg} successfully!")
+        await ctx.reply(embed=embed)
+
+    @staticmethod
+    def create_token() -> int:
+        secret = pyotp.random_base32()
+        totp = pyotp.TOTP(secret)
+        return int(totp.now())
+
+    @staticmethod
+    async def check_delete_channel(voice_channel: VoiceChannel, db: Union[GamingChannels, StudyChannels],
+                                   logger: logging.Logger,
+                                   reset_delete_at: tuple[bool, PrimitiveMongoData] = (False, None)) -> bool:
+        if len({member for member in voice_channel.members if not member.bot}) == 0:
+            document: list[Union[StudyChannel, GamingChannel]] = await db.find(
+                {DBKeyWrapperEnum.VOICE.value: voice_channel.id})
+
+            if not document:
+                logger.error(voice_channel.name)
+                raise DatabaseIllegalState
+
+            document: Union[StudyChannel, GamingChannel] = document[0]
+
+            if type(document) == GamingChannel or not document.deleteAt or (
+                    not reset_delete_at[0] and datetime.now() > document.deleteAt):
+                await document.voice.delete(reason="No longer used")
```

Comment:
Here, errors could occur due to permissions or already deleted chats.
We should deal with that possible issue.

Reply:
fixed if already deleted, but if there can't be an error because of permissions since the bot has admin role